RecoBTag/ONNXRuntime/plugins/DeepCombinedONNXJetTagsProducer.cc

```diff
@@ -0,0 +1,345 @@
+                                      "input_11",
+                                      "input_12"});
+  desc.add<edm::FileInPath>("model_path",
+                            edm::FileInPath("RecoBTag/Combined/data/DeepVertex/phase1_deepvertexcombined.onnx"));
+  desc.add<std::vector<std::string>>("output_names", {"dense_13"});
+  desc.add<std::vector<std::string>>("flav_names", std::vector<std::string>{"probb", "probc", "probuds", "probg"});
+  desc.add<double>("min_jet_pt", 15.0);
+  desc.add<double>("max_jet_eta", 2.5);
+
+  descriptions.add("pfDeepCombinedJetTags", desc);
+}
+
+std::unique_ptr<ONNXRuntime> DeepCombinedONNXJetTagsProducer::initializeGlobalCache(const edm::ParameterSet& iConfig) {
+  return std::make_unique<ONNXRuntime>(iConfig.getParameter<edm::FileInPath>("model_path").fullPath());
+}
+
+void DeepCombinedONNXJetTagsProducer::globalEndJob(const ONNXRuntime* cache) {}
+
+void DeepCombinedONNXJetTagsProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  edm::Handle<TagInfoCollection> tag_infos;
+  iEvent.getByToken(src_, tag_infos);
+
+  data_.clear();
+
+  std::vector<std::unique_ptr<JetTagCollection>> output_tags;
+  if (!tag_infos->empty()) {
+    unsigned int good_taginfo_count = 0;
+    for (unsigned jet_n = 0; jet_n < tag_infos->size(); ++jet_n) {
+      const auto& jet_ref = (*tag_infos)[jet_n].jet();
+      if (jet_ref->pt() > min_jet_pt_ && std::fabs(jet_ref->eta()) < max_jet_eta_)
+        good_taginfo_count++;
+    }
+
+    // init data storage w correct size
+    for (const auto& len : input_sizes_) {
+      data_.emplace_back(good_taginfo_count * len, 0);
+    }
+
+    // initialize output collection
+    auto jet_ref = tag_infos->begin()->jet();
+    auto ref2prod = edm::makeRefToBaseProdFrom(jet_ref, iEvent);
+    for (std::size_t i = 0; i < flav_names_.size(); i++) {
+      output_tags.emplace_back(std::make_unique<JetTagCollection>(ref2prod));
+    }
+
+    // convert inputs
+    unsigned int inputs_done_count = 0;
+    for (unsigned jet_n = 0; jet_n < tag_infos->size(); ++jet_n) {
+      const auto& jet_ref = (*tag_infos)[jet_n].jet();
+      if (jet_ref->pt() > min_jet_pt_ && std::fabs(jet_ref->eta()) < max_jet_eta_) {
+        const auto& taginfo = (*tag_infos)[jet_n];
+        make_inputs(inputs_done_count, taginfo);
+        inputs_done_count++;
+      }
+    }
+
+    // run prediction
+    assert(inputs_done_count == good_taginfo_count);
+    const auto outputs = globalCache()->run(input_names_, data_, {}, output_names_, good_taginfo_count)[0];
+    assert(outputs.size() == flav_names_.size() * good_taginfo_count);
+
+    // get the outputs
+    unsigned i_output = 0;
+    for (unsigned jet_n = 0; jet_n < tag_infos->size(); ++jet_n) {
+      const auto& jet_ref = (*tag_infos)[jet_n].jet();
+      for (std::size_t flav_n = 0; flav_n < flav_names_.size(); flav_n++) {
+        if (jet_ref->pt() > min_jet_pt_ && std::fabs(jet_ref->eta()) < max_jet_eta_) {
```

Comment:
here you compute the jet cut the 3rd time, as suggested elsewhere, save the bool cut in a vector to prevent reimplementation.